subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -196,38 +257,24 @@ int nrf_cloud_fota_init(nrf_cloud_fota_callback_t cb)
 		fota_dl_initialized = true;
 	}
 
-	ret = settings_load_subtree(settings_handler_fota.name);
-	if (ret) {
-		LOG_ERR("Cannot load settings: %d", ret);
-		return ret;
-	}
-
-	ret = nrf_cloud_pending_fota_job_process(&saved_job, &reboot_required);
-
-	if (ret == 0) {
-
-		/* save status and update when cloud connection is ready */
-		save_validate_status(saved_job.id, saved_job.type, saved_job.validate);
-
-		if (reboot_required) {
-			LOG_INF("Rebooting to complete FOTA update...");
-			sys_reboot(SYS_REBOOT_COLD);
-		}
+	ret = nrf_cloud_fota_pending_job_validate(NULL);
```

Comment:
Due to the change in the meaning of the return value in `nrf_cloud_fota_pending_job_validate` and in `nrf_cloud_fota_init`, setting `ret` directly like this makes the logic hard to follow, even though it is technically less verbose

Perhaps declare a separate variable to receive this value from `nrf_cloud_fota_pending_job_validate` and set `ret` accordingly?

Reply:
added comment to explain `ret`